crates/bevy_ecs/src/reflect.rs

```diff
@@ -30,6 +33,18 @@ impl ReflectComponent {
         (self.reflect_component)(world, entity)
     }
 
+    /// # Safety
+    /// This method does not prevent you from having two mutable pointers to the same data, violating Rust's aliasing rules. To avoid this:
+    /// * Only call this method in a thread-local system to avoid sharing across threads.
+    /// * Don't call this method more than once in the same scope for a given component.
```

Comment:
Could we have a safe variant of this function that takes `&mut World`?

Reply:
ooh i like that. good call!